### PR TITLE
Only allow project drag and drop when no dialog open

### PR DESCRIPTION
**Pull request by microbit-grace**
Also, tweak displaying of dialogs to ensure no overlapping of connection dialog when other dialogs are up

Related task: https://microbit-global.monday.com/boards/1550536443/pulses/1656281174

### CODE REVIEW

File: src/components/DefaultPageLayout.tsx [hunk 1]
```diff
@@ -85,60 +97,71 @@ const DefaultPageLayout = ({
...
-      {!isEditorOpen && stage === TrainModelDialogStage.Closed && (
-        <ConnectionDialogs />
-      )}
+      {!isEditorOpen &&
+        isTrainDialogClosed &&
+        isTourClosed &&
+        isSaveDialogClosed && <ConnectionDialogs />}
       {!isEditorOpen && <TrainModelDialogs />}
-      {!isEditorOpen && stage === TrainModelDialogStage.Closed && <Tour />}
+      {!isEditorOpen && <Tour />}
```
Comment (microbit-matt-hillsdon) [inline src/components/DefaultPageLayout.tsx:116]: This approach kinda feels like it's reached its limit. One to revisit when we've consolidated the state.

Comment (microbit-grace) [reply src/components/DefaultPageLayout.tsx:116]: Agreed. That would be quite satisfying to consolidate

Comment (microbit-matt-hillsdon) [reply src/components/DefaultPageLayout.tsx:116]: Was it not viable to move the project drop overlay to a z-index below the modal dialog overlay?

Comment (microbit-grace) [reply src/components/DefaultPageLayout.tsx:116]: Haven't tried that. Will give that a go

Comment (microbit-grace) [reply src/components/DefaultPageLayout.tsx:116]: Seems like the part detecting the drag over doesn't have a zIndex (https://github.com/microbit-foundation/ml-trainer/blob/drop-target-when-needed/src/components/FileDropTarget.tsx#L47).

I also tried changing the z-Index of the project drop overlay. The onDragOver is still triggered, whilst the  onDragLeave does not get triggered. The result is that the overlay is stuck on screen.

**Pull request merged**